Force user to select a sample

naegelyd opened the pull request:
Fix #64.

This has the side effect of blocking all context changes until the sample is selected but that is necessary given the way required concepts are implemented currently in Cilantro. There is a notification made to the user when they try to modify the context with no sample selected so it should be clear enough that no one is confused.

naegelyd committed:
Force user to select a sample

This has the side effect of blocking all context changes until the sample is selected but that is necessary given the way required concepts are implemented currently in Cilantro. There is a notification made to the user when they try to modify the context with no sample selected so it should be clear enough that no one is confused.

bruth commented on varify/static/scripts/coffeescript/main.coffee:
```diff
@@ -62,6 +62,18 @@ require
+    # Mark the Sample concept as required and display a notification to the
+    # user when it is not populated.
+    c.config.set('query.concepts.required', [2])
+    c.on c.CONTEXT_INVALID, (concepts) ->
```
There is a `c.CONTEXT_REQUIRED` event as well when the sample is attempted to be removed.

naegelyd committed:
Update Cilantro to 2.2.11

naegelyd committed:
Fix custom controls config options to match changes in Cilantro 2.2.11

naegelyd committed:
Notify user on both CONTEXT_REQUIRED and _INVALID events

This was not working pre Cilantro 2.2.11 because CONTEXT_REQUIRED did not supply the correct argument to handlers.
